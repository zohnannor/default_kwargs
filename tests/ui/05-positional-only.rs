#![warn(clippy::all, clippy::pedantic, clippy::nursery, rustdoc::all)]

use default_kwargs::{default_args, keyword_args};

default_args! {
    fn thing(x: i32, y: i32) -> (i32, i32) {
        (x, y)
    }
}

fn main() {
    let (r, s) = keyword_args! { thing(y = 666) };
}
