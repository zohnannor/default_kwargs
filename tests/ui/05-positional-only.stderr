error[E0061]: this function takes 3 arguments but 1 argument was supplied
  --> tests/ui/05-positional-only.rs:12:34
   |
12 |     let (r, s) = keyword_args! { thing(y = 666) };
   |                  ----------------^^^^^-----------
   |                  |               |
   |                  |               expected 3 arguments
   |                  supplied 1 argument
   |
note: function defined here
  --> tests/ui/05-positional-only.rs:6:8
   |
5  |   default_args! {
   |  _-
   | |_|
   | |
6  | |     fn thing(x: i32, y: i32) -> (i32, i32) {
   | |        ^^^^^ ------
7  | |         (x, y)
8  | |     }
9  | | }
   | |_-
   | |_|
   |

error[E0560]: struct `ThingArgs` has no field named `y`
  --> tests/ui/05-positional-only.rs:12:40
   |
12 |     let (r, s) = keyword_args! { thing(y = 666) };
   |                                        ^ `ThingArgs` does not have this field
